packages/redux-query-react/src/hooks/use-request-on-demand.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+// NOTE(kelson): The public flow type interface for this hook is defined in use-request-on-demand.js.flow
+
+import * as React from 'react';
+import { useDispatch } from 'react-redux';
+import { requestAsync } from 'redux-query';
+import type { ActionPromiseValue, QueryConfig } from 'redux-query/types.js.flow';
+
+import type { QueryState } from '../types';
+import useQueryState from './use-query-state';
```

Comment:
It might be nice to also export `useQueryState` as an escape hatch for use in unforeseen problems. Ultimately it's the lack of that info that prevents this hook from being recreated on the frontend

Reply:
If you export something in an open source package that's implicitly committing to support it as part of the API.

So if you were to do this you should either be comfortable doing that or clearly flag it as an unstable API.